integration_test/pg/storage_test.exs

```diff
@@ -147,6 +147,69 @@ defmodule Ecto.Integration.StorageTest do
     assert contents =~ ~s[INSERT INTO public."schema_migrations" (version) VALUES]
   end
 
+  test "when :dump_prefixes is not provided, structure is dumped for all schemas but only public schema migration records are inserted" do
+    # Create the test_schema schema
+    run_psql(~s[CREATE SCHEMA test_schema], [PoolRepo.config()[:database]])
+
+    # Run migrations
+    version = @base_migration + System.unique_integer([:positive])
+    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false)
+    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false, prefix: "test_schema")
+
+    {:ok, path} = Postgres.structure_dump(tmp_path(), TestRepo.config())
+    contents = File.read!(path)
+
+    assert contents =~ "CREATE TABLE public.schema_migrations"
+    assert contents =~ ~s[INSERT INTO public."schema_migrations" (version) VALUES (#{version})]
+    assert contents =~ "CREATE TABLE test_schema.schema_migrations"
+    refute contents =~ ~s[INSERT INTO test_schema."schema_migrations" (version) VALUES (#{version})]
+  after
+    run_psql(~s[DROP SCHEMA test_schema], [PoolRepo.config()[:database]])
+  end
```

Comment:
```suggestion
  def create_schema(database) do
    run_psql(~s[CREATE SCHEMA test_schema], [database])
  end

  def create_schema_migrations(database) do
    run_psql(~s[CREATE TABLE test_schema.schema_migrations], [database])
  end

  def drop_schema(database) do
    run_psql(~s[DROP SCHEMA test_schema], [database])
  end

  def drop_schema_migrations(database) do
    run_psql(~s[DROP TABLE test_schema.schema_migrations], [database])
  end
  
  test "when :dump_prefixes is not provided, structure is dumped for all schemas but only public schema migration records are inserted" do
    # Create the test_schema schema
    create_schema(PoolRepo.config()[:database])

    # Run migrations
    version = @base_migration + System.unique_integer([:positive])
    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false)
    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false, prefix: "test_schema")

    {:ok, path} = Postgres.structure_dump(tmp_path(), TestRepo.config())
    contents = File.read!(path)

    assert contents =~ "CREATE TABLE public.schema_migrations"
    assert contents =~ ~s[INSERT INTO public."schema_migrations" (version) VALUES (#{version})]
    assert contents =~ "CREATE TABLE test_schema.schema_migrations"
    refute contents =~ ~s[INSERT INTO test_schema."schema_migrations" (version) VALUES (#{version})]
  after
    drop_schema_migrations(PoolRepo.config()[:database])
    drop_schema(PoolRepo.config()[:database])
  end
```

- Since we have a few tests now it's a bit easier to create functions, similar to `create_database`, `create_posts`, etc. We can put these new functions at the top with the other ones and use them in the 3 tests.
- Also we have to drop the table we created first otherwise when we drop the schema it says error can't drop because the table is depending on it